src/labelformat/formats/yolov8.py

```diff
@@ -50,28 +50,75 @@ def __init__(self, input_file: Path, input_split: str) -> None:
             )
 
     def get_categories(self) -> Iterable[Category]:
-        for category_id, category_name in self._config_data["names"].items():
-            yield Category(
-                id=int(category_id),
-                name=category_name,
+        """Get categories from YOLOv8 config file.
+
+        Supports two formats for the 'names' field:
+        - dictionary mapping id to name
+          Example:
+          names:
+            0: person
+            1: dog
+            2: cat
+        - list of names where index is the id
+          Example:
+          names:
+            - person
+            - dog
+            - cat
+          or
+          names: [person, dog, cat]
+        Returns:
+            Iterator of Category objects.
+        """
+        names = self._config_data["names"]
+        if isinstance(names, dict):
```

Comment:
Did you consider just using a for loop? Then you can get rid of the whole if-else clause:
```python
names = self._config_data["names"]
for category_id in range(len(names)):
    yield Category(
        id=category_id,
        name=names[category_id],
    )
```

The only disadvantage is that the keys in dictionary format must always be zero-based indices. I think based on the format this should be the case. You could handle the exception by catching a `KeyError` and then re-raising.